### ISSUE TITLE
Unhandled exception on Autosubmit: time data '24:00' does not match format '%H:%M'

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#539]
In GitLab by @mcastril on Jun 4, 2020, 12:31

Hi. @ mariag is having problems with the wrapper. 

```
[[31m[CRITICAL] Unhandled exception on Autosubmit: time data '24:00' does not match format '%H:%M'
Traceback (most recent call last):
  File "/shared/earth/software/autosubmit/3.12.0-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.12.0-py2.7.egg/autosubmit/autosubmit.py", line 416, in parse_args
    return Autosubmit.run_experiment(args.expid, args.notransitive)
  File "/shared/earth/software/autosubmit/3.12.0-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.12.0-py2.7.egg/autosubmit/autosubmit.py", line 1034, in run_experiment
    wrapper_job.check_status(status)
  File "/shared/earth/software/autosubmit/3.12.0-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.12.0-py2.7.egg/autosubmit/job/job.py", line 1005, in check_status
    self.check_inner_job_status()
  File "/shared/earth/software/autosubmit/3.12.0-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.12.0-py2.7.egg/autosubmit/job/job.py", line 1009, in check_inner_job_status
    self.check_inner_jobs_completed(self.running_jobs_start.keys())
  File "/shared/earth/software/autosubmit/3.12.0-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.12.0-py2.7.egg/autosubmit/job/job.py", line 1027, in check_inner_jobs_completed
    self._check_inner_job_wallclock(job)
 File "/shared/earth/software/autosubmit/3.12.0-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.12.0-py2.7.egg/autosubmit/job/job.py", line 1120, in _is_over_wallclock
    wallclock = datetime.datetime.strptime(wallclock, '%H:%M')
  File "/shared/earth/software/Python/2.7.9-foss-2015a/lib/python2.7/_strptime.py", line 325, in _strptime
    (data_string, format))
ValueError: time data '24:00' does not match format '%H:%M'
```

This seems to be a Python problem:

https://stackoverflow.com/questions/25132870/valueerror-time-data-2400-does-not-match-format-hm

On the other hand I find it strange because I would swear that we have used 24:00 in other workflows.

@dbeltrankyl @wuruchi

**Comment by kinow** [comment]
In GitLab by @mcastril on Jun 4, 2020, 12:32

mentioned in issue auto-ecearth3#1304

**Comment by kinow** [comment]
In GitLab by @mcastril on Jun 4, 2020, 12:32

@pablohe did you try this format in the testing suite?

**Comment by kinow** [comment]
In GitLab by @pablohe on Jun 4, 2020, 12:37

I didn't a 24 hs wallclock is not common

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jun 4, 2020, 12:40

Hi, 

I guess "24" does not exist as a time ( at least the clock never reach 24 it goes to 00:00).

So in order to put 1 day, in the current code and as a dirty solution, put 24:01 or 23:59. 

Another way to do it should be modifying the AS code to parse time > 24:00 in day-hour-minutes

**Comment by kinow** [comment]
In GitLab by @mcastril on Jun 4, 2020, 12:45

> So in order to put 1 day, in the current code and as a dirty solution, put 24:01 or 23:59. 

I was sure about 23:59 but not about 24:01. It is strange because the clock never arrives to 24:01 too. But I guess it will work from that number on wards.

> Another way to do it should be modifying the AS code to parse time > 24:00 in day-hour-minutes 

I think so. Not for 3.12.0 but we should do it in 3.13.0 because from our point of view we are not talking about day times but accumulated hours:minutes.

**Comment by kinow** [comment]
In GitLab by @pablohe on Jun 4, 2020, 12:45

Maybe you can use python timedelta, a wallclock exactly this.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jun 4, 2020, 12:48

> I was sure about 23:59 but not about 24:01. It is strange because the clock never arrives at 24:01 too. But I guess it will work from that number onwards.

Maybe internally, it just parses 24:01 as another day + 01 minutes but not for 24:00. Although it would happen also for 48 and 72, and I think I put 72 sometimes without issues but not sure right now

**Comment by kinow** [comment]
In GitLab by @mcastril on Jun 8, 2020, 15:12

I think this is a good idea.

**Comment by kinow** [comment]
In GitLab by @ mariag on Jun 9, 2020, 11:28

FYI: 23:59 works fine.

**Comment by kinow** [comment]
In GitLab by @mcastril on Aug 10, 2020, 13:03

changed due date to August 21, 2020

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 13, 2020, 13:16

Hi, 

This was already implemented and uploaded into 3.13.0b , if you encounter any issue, please report it back

**Comment by kinow** [comment]
In GitLab by @mcastril on Nov 2, 2020, 13:35

closed
